lib/rubygems/test_utilities.rb

```diff
@@ -54,10 +54,11 @@ def find_data(path, nargs = 3)
       raise Gem::RemoteFetcher::FetchError.new("no data for #{path}", path)
     end
 
-    data = @data[path]
-
-    data.flatten! and return data.shift(nargs) if data.respond_to?(:flatten!)
-    data
+    if @data[path].kind_of?(Array) && @data[path].first.kind_of?(Array)
+      @data[path].shift
+    else
+      @data[path]
+    end
```

Comment:
this was needed so that `@fetcher.data["#{@host}/api/v1/gems"]` can return more than two responses. `find_data` seems a bit overloaded. `data.flatten! and return data.shift(nargs) if data.respond_to?(:flatten!)` was my previous last-ditch effort of supporting more than one response from faker. does this need something more elegant?

Reply:
I think we could improve it in the future but what you did here is more flexible and no less elegant than the previous solution, so I'm +1 on what you did.